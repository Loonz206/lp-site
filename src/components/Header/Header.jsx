import React from "react";
import { Link } from "gatsby";
import PropTypes from "prop-types";
import "./Header.scss";

const Header = props => {
  // eslint-disable-next-line react/prop-types
  const { links, handleClick, state } = props;

  const renderLinks = links.map(({ path, name, id }) => (
    <li key={id}>
      <Link to={path} className="nav-link" activeClassName="active">
        {name}
      </Link>
    </li>
  ));

  return (
    <header>
      <a href="#content" className="sr-only">
        Skip to Content
      </a>
      <a
        href="#menu"
        className={state === "hello" ? "menu-link" : "menu-link active"}
        onClick={() => handleClick()}
      >
        Menu &#x2630;
      </a>
      <nav id="menu" role="navigation">
        <ul>{renderLinks}</ul>
      </nav>
    </header>
  );
};

export default Header;

Header.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape()).isRequired
};
