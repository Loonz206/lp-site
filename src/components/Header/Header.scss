@import '../../styles/variables.scss';

// active class for navigation
.nav-link.active {
  color: $primaryColor;
  background-color: $white;
}

.menu-link {
  text-decoration: none;
  color: $primaryText;

  &:hover,
  &:focus {
    color: $primaryColor;
  }

  &:active {
    background-color: $accentColor;
    color: $white;
  }
}

.wrap {
  position: relative;
  -webkit-transition: all 0.3s ease-out;
  -moz-transition: all 0.3s ease-out;
  -ms-transition: all 0.3s ease-out;
  -o-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
  right: 0;
}

.wrap.active {
  right: 16em; //right based
}

a.menu-link {
  float: right; //right based
  display: block;
  padding: 1em;
  border: none;
}

nav[role=navigation] {
  background: $primaryColor;
  clear: both;
  overflow: hidden;
}

.js nav[role=navigation] {
  width: 16em;
  height: 100%;
  position: absolute;
  top: 0;
  right: -16em; //right based
  overflow-x: hidden;
}

nav[role=navigation] ul {
  margin: 0;
  padding: 0;
  border-top: 1px solid $primaryColor;
}

nav[role=navigation] li a {
  display: block;
  padding: 0.8em;
  color: $white;
  border-bottom: 1px solid $white;
  text-decoration: none;
  transition: .5s;

  &:hover,
  &:focus {
    background-color: rgba(255, 255, 255, .5);
  }

  &:active {
    background-color: $accentColor;
  }
}

.main {
  clear: both;
  width: 94%;
  max-width: 42em;
  margin: 80px auto;
}

//big screen
@media screen and (min-width: 41.25em) {

  a.menu-link {
    display: none;
  }

  .js nav[role=navigation] {
    max-width: none;
    position: static;
    width: auto;
  }

  .wrap.active {
    right: 0; //right based fly out
  }

  nav[role=navigation] ul {
    margin: 0;
    border: 0;
    float: right; //The navigation flows on the right
  }

  nav[role=navigation] li {
    display: inline-block;
    margin: 0 0.25em;
  }

  nav[role=navigation] li a {
    border: 0;
  }
}
