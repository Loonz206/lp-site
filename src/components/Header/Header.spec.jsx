import React from "react";
import renderer from "react-test-renderer";

import Header from "./Header";

describe("Header", () => {
  it("renders correctly", () => {
    const links = [
      {
        name: "hello",
        path: "/hello",
        id: "4"
      }
    ];
    const tree = renderer.create(<Header links={links} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
// testin
