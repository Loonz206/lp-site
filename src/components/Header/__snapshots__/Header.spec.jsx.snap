// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <a
    className="sr-only"
    href="#content"
  >
    Skip to Content
  </a>
  <a
    className="menu-link active"
    href="#menu"
    onClick={[Function]}
  >
    Menu â˜°
  </a>
  <nav
    id="menu"
    role="navigation"
  >
    <ul>
      <li>
        <a
          className="nav-link"
          href="/hello"
        >
          hello
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
